src/components/candevice/candevicemodel.cpp

```diff
@@ -0,0 +1,98 @@
+#include "candevicemodel.h"
+#include <QtCore/QDir>
+#include <QtCore/QEvent>
+
+#include <QtWidgets/QFileDialog>
+
+#include <nodes/DataModelRegistry>
+
+#include "datamodeltypes/candevicedata.h"
+
+CanDeviceModel::CanDeviceModel()
+    : label(new QLabel())
+{
+    label->setAlignment(Qt::AlignVCenter | Qt::AlignHCenter);
+
+    label->setFixedSize(75, 25);
+
+    label->installEventFilter(this);
+
+    label->setAttribute(Qt::WA_TranslucentBackground);
+
+    canDevice = new CanDevice(factory);
```

Comment:
Passing factory not needed. Use empty constructor to use default one. I will refactor it again shortly to change the name to context as it seems to be more relevant here...